commit a6305580b27f0bf1d268b5a3531c8332a57c3864
Author: Kevin Kofler <Kevin@tigcc.ticalc.org>
Date:   Sun Nov 25 16:20:00 2012 +0000

    Fix moc from choking on boost headers.
    
    Fixes several build failures with boost >= 1.48, for example
    the kdepim build: https://bugs.kde.org/show_bug.cgi?id=304111
    
    Task-number: QTBUG-22829
    Change-Id: Ic37a5290955fbfd18894a228cc9d57615edddea6
    Reviewed-by: Olivier Goffart <ogoffart@woboq.com>

diff --git a/src/tools/moc/main.cpp b/src/tools/moc/main.cpp
index 33344fc..f558a5e 100644
--- a/src/tools/moc/main.cpp
+++ b/src/tools/moc/main.cpp
@@ -187,6 +187,10 @@ int runMoc(int _argc, char **_argv)
     Moc moc;
     pp.macros["Q_MOC_RUN"];
     pp.macros["__cplusplus"];
+
+    // Workaround a bug while parsing the boost/type_traits/has_operator.hpp header. See QTBUG-22829
+    pp.macros["BOOST_TT_HAS_OPERATOR_HPP_INCLUDED"];
+
     QByteArray filename;
     QByteArray output;
     FILE *in = 0;
